tests/unit_tests/test_transaction.py

```diff
@@ -1414,6 +1414,45 @@ def test_application_delete(self):
         self.assertEqual(i, call)
 
 
+class TestHeartbeatTransactions(unittest.TestCase):
+    sender = "7ZUECA7HFLZTXENRV24SHLU4AVPUTMTTDUFUBNBD64C73F3UHRTHAIOF6Q"
+    genesis = "JgsgCaCTqIaLeVhyL6XlRu3n7Rfk2FxMeK+wRSaQ7dI="
+    hb_address = "MTTDUFUBNBD64C73F3UHRTHAIOF6Q7ZUECA7HFLZTXENRV24SHLU4AVPUT"
+    hb_proof = {
+        "Sig": "AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==",
+        "PK": "AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==",
+        "PK2": "AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==",
+        "PK1Sig": "AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==",
+        "Pk2Sig": "AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==",
+    }
+    hb_seed = "XNOEe4VHUzo+8kWmV8ZE1T/GViVIkC9zbz0IhwhKTjY="
+    hb_vote_id = "XNOEe4VHUzo+8kWmV8ZE1T/GViVIkC9zbz0IhwhKTjY="
+    hb_key_dilution = 10000
+    note = b"\x00"
+    lease = b"\0" * 32
+    params = transaction.SuggestedParams(0, 1, 100, genesis)
+
+    def test_heartbeat(self):
+        # Test serializing a basic heartbeat with all standard fields filled in (save for rekey).
+        hb = transaction.HeartbeatTxn(
+            self.sender,
+            self.params,
+            self.hb_address,
+            self.hb_proof,
+            self.hb_seed,
+            self.hb_vote_id,
+            self.hb_key_dilution,
+            self.note,
+            self.lease,
+        )
+
+        enc = encoding.msgpack_encode(hb)
+        re_enc = encoding.msgpack_encode(encoding.msgpack_decode(enc))
+        self.assertEqual(enc, re_enc)
+
+        self.assertEqual(transaction.HeartbeatTxn.undictify(hb.dictify()), hb)
```

Comment:
we used to have so-called "golden" tests - txn created by algod in base64 decoded in a test and some fields compared, serialized back and compared back to golden

Reply:
I was following the pattern already here. I think I see some of those golden tests in indexer, thought it was message-pack encoded?

Reply:
I guess I'll add sdk-testing test case as a followup